<launch>
  <arg name="camera_name"/>
  <arg name="pkg_name" value="foscam_ros"/>
  <arg name="manager" value="camera_manager"/>

  <arg name="decoder" default="true"/>
  <arg name="crop" default="true"/>
  <arg name="rectify" default="$(optenv RECTIFY true)"/>
  <arg name="param_file_name" default="default"/>

  <arg name="config_dir" default="$(find foscam_ros)/config"/>
  <arg name="crop_param_file" default="$(arg config_dir)/crop_node/$(arg param_file_name).yaml"/>
  <arg name="camera_param_file" default="$(arg config_dir)/foscam_ros_node/$(arg param_file_name).yaml"/>


  <group ns="$(arg camera_name)">

    <node name="camera_node" pkg="$(arg pkg_name)" type="foscam_ros_node.py" output="screen" required="true">
      <param name="camera_name" value="$(arg camera_name)"/>
      <rosparam command="load" file="$(arg camera_param_file)"/>
    </node>

    <node if="$(arg decoder)" name="decoder" type="republish" pkg="image_transport" output="screen"
      args="compressed in:=camera_node/image raw out:=camera_node/image" />

    <node pkg="nodelet" type="nodelet" name="$(arg manager)"  args="manager"/>

    <node if="$(arg crop)" pkg="nodelet" type="nodelet" args="load image_proc/crop_decimate $(arg manager)" name="crop">
      <rosparam command="load" file="$(arg crop_param_file)"/>
      <!-- remap input topics -->
      <remap from="camera/image_raw" to="camera_node/image" />
      <remap from="camera/camera_info" to="camera_node/camera_info" />
      <!-- remap output topics -->
      <remap from="camera_out/image_raw" to="camera_node/crop/image" />
      <remap from="camera_out/camera_info" to="camera_node/crop/camera_info" />
    </node>

    <node if="$(arg rectify)" pkg="nodelet" type="nodelet" args="load image_proc/rectify $(arg manager)" name="rectifier">
      <!-- remap input topics -->
      <remap from="image_mono" to="camera_node/crop/image" />
      <remap from="camera_info" to="camera_node/crop/camera_info" />
      <!-- remap output topics -->
      <remap from="image_rect" to="camera_node/crop/rect/image" />
    </node>

    <node if="$(arg rectify)" name="rect_info_relay" type="relay" pkg="topic_tools" output="screen"
      args="camera_node/crop/camera_info camera_node/crop/rect/camera_info" />

  </group>

</launch>
